terraform/modules/daos_server/templates/daos_startup_script.tftpl

```diff
@@ -8,30 +8,47 @@ DAOS_MOUNT_DIR="/var/daos"
 fetch_attr()
 {
   local attr_name=$*
-  curl -s ${METADATA_URL}/${attr_name} -H "Metadata-Flavor: Google"
+  curl -s $${METADATA_URL}/$${attr_name} -H "Metadata-Flavor: Google"
 }
 
 echo "BEGIN: Setting up DAOS server"
 
 systemctl stop daos_server
 
 # Create server config files
-mkdir -p "${DAOS_CONFIG_DIR}"
-cd "${DAOS_CONFIG_DIR}"
+mkdir -p "$${DAOS_CONFIG_DIR}"
+# Create directory for engine logs and tmpfs mount point
+mkdir -p "$${DAOS_MOUNT_DIR}"
+cp /tmp/configure_daos.sh  $${DAOS_MOUNT_DIR}/configure_daos.sh
+cd "$${DAOS_CONFIG_DIR}"
+
 fetch_attr "daos_server_yaml_content" > daos_server.yml
 fetch_attr "daos_control_yaml_content" > daos_control.yml
 fetch_attr "daos_agent_yaml_content" > daos_agent.yml
 
-# Create directory for engine logs and tmpfs mount point
-mkdir -p "${DAOS_MOUNT_DIR}"
-
 # Modify systemd script for GCP
 # First, run daos_server as root since GCP does not support VFIO
-sed -i "s/User=daos_server/User=root/; s/Group=daos_server/Group=root/" ${DAOS_SERVER_SYSTEMD_FILE}
+sed -i "s/User=daos_server/User=root/; s/Group=daos_server/Group=root/" $${DAOS_SERVER_SYSTEMD_FILE}
 
 # enable daos_server in systemd (will be started automatically at boot time)
 systemctl enable daos_server
 
 systemctl start daos_server
 
+if [ `hostname` != "daos-server-0001" ]; then
```

Comment:
same comment about the instance_base_name.  But since you render this here, you may directly render the variable value too and pass it to the configure_daos.sh